notebooks/05a Modelling - regression.ipynb

```diff
@@ -0,0 +1,1001 @@
+    "    estimator=RandomForestRegressor(),\n",
+    "    param_grid={},\n",
+    "    cv=5,\n",
+    "    scoring=\"neg_mean_squared_error\",\n",
+    "    verbose=1,\n",
+    "    n_jobs=-1,\n",
+    "    refit=True,\n",
+    ")\n",
+    "\n",
+    "# takes ~7 mins to run on a STANDARD_D13_V2\n",
+    "models[model_name] = train_model(gsc, X_train, y_train)\n",
+    "models[model_name]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Catboost\n",
+    "\n",
+    "Boosted tree optimised for categorical features"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "gather": {
+     "logged": 1653320819294
+    }
+   },
+   "outputs": [],
+   "source": [
+    "model_name = \"catboost\"\n",
+    "\n",
+    "# extract categorical features\n",
+    "num_features = [\n",
+    "    \"AGE_ON_ADMISSION\",\n",
+    "    \"EL CountLast12m\",\n",
+    "    \"EMCountLast12m\",\n",
+    "    \"OP First CountLast12m\",\n",
+    "    \"OP FU CountLast12m\",\n",
+    "]\n",
+    "cat_features = list(set(X_train_catboost.columns) - set(num_features))\n",
+    "\n",
+    "gsc = GridSearchCV(\n",
+    "    estimator=CatBoostRegressor(verbose=False, cat_features=cat_features),\n",
+    "    param_grid={},\n",
+    "    cv=5,\n",
+    "    scoring=\"neg_mean_squared_error\",\n",
+    "    verbose=1,\n",
+    "    n_jobs=-1,\n",
+    "    refit=True,\n",
+    ")\n",
+    "\n",
+    "# takes ~4 mins to run on a STANDARD_D13_V2\n",
+    "models[model_name] = train_model(gsc, X_train_catboost, y_train)\n",
+    "models[model_name]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### XGBoost"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "gather": {
+     "logged": 1653320996266
+    },
+    "jupyter": {
+     "outputs_hidden": false,
+     "source_hidden": false
+    },
+    "nteract": {
+     "transient": {
+      "deleting": false
+     }
+    }
+   },
+   "outputs": [],
+   "source": [
+    "model_name = \"xgboost\"\n",
+    "\n",
+    "# define gridsearch parameters\n",
+    "gsc = GridSearchCV(\n",
+    "    estimator=XGBRegressor(random_state=42),\n",
+    "    param_grid={},\n",
+    "    cv=5,\n",
+    "    scoring=\"neg_mean_squared_error\",\n",
+    "    verbose=1,\n",
+    "    n_jobs=-1,\n",
+    "    refit=True,\n",
+    ")\n",
+    "\n",
+    "# takes ~2 mins to run on a STANDARD_D13_V2\n",
+    "models[model_name] = train_model(gsc, X_train, y_train)\n",
+    "models[model_name]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Save models"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "gather": {
+     "logged": 1653387928947
+    }
+   },
+   "outputs": [],
+   "source": [
+    "# save models outside the git tree\n",
+    "with open(\"../../models/regression.pickle\", \"wb\") as handle:\n",
```

Comment:
models folder is missing and can be stored within the repo.

Reply:
We would like to exclude model artefacts from the repository as they may contain information on the training data that we are not in a position to share publicly.